src/Amazon.php

```diff
@@ -25,13 +32,105 @@ public static function woocommerce_amazon_pa_update_checkout_session_payload($pa
       return defined('WC_AMAZON_PAY_VERSION') && version_compare(WC_AMAZON_PAY_VERSION, '2.0', '>=');
     }
-    return false;
+    return FALSE;
+  }
+
+  /**
+   * Provides shipping method ID to wp-lister-amazon plugin filter.
+   *
+   * @implements wpla_shipping_service_id_map
+   */
+  public static function wpla_shipping_service_id_map($map, $order) {
+    $available_method = self::get_available_shipping_methods($order);
+
+    if ($available_method) {
+      $id = $available_method->get_method_id();
+      return [
+        'Std DE Dom' => $id,
+        'Std DE Dom_2' => $id,
+        'AD DE Dom' => $id,
+        'MFN AD Intl' => $id,
+      ];
+    }
+    return $map;
+  }
+
+  /**
+   * Provides shipping method title to wp-lister-amazon plugin filter.
+   *
+   * @implements wpla_shipping_service_title_map
+   */
+  public static function wpla_shipping_service_title_map($map, $order) {
+    $available_method = self::get_available_shipping_methods($order);
+
+    if ($available_method) {
+      $label = $available_method->get_label();
+      return [
+        'Std DE Dom' => $label,
+        'Std DE Dom_2' => $label,
+        'AD DE Dom' => $label,
+        'MFN AD Intl' => $label,
+      ];
+    }
+    return $map;
+  }
+
+  /**
+   * Recreates WC_Cart instance and returns the first available shipping method.
+   *
+   * @param WC_Order $order
+   *   Woocommerce order to calculate its shipping method.
+   *
+   * @return WC_Shipping_Rate|bool
+   *   The first available shipping method or FALSE.
+   */
+  public static function get_available_shipping_methods(\WC_Order $order) {
+
+    if (isset(self::$firstAvailableMethod[$order->get_id()])) {
+      return self::$firstAvailableMethod[$order->get_id()];
+    }
+
+    $country = $order->get_shipping_country() ?? NULL;
+    $postcode = $order->get_shipping_postcode() ?? NULL;
+    $state = $order->get_shipping_state() ?? '';
+    $city = $order->get_shipping_city() ?? '';
+
+    $order_items = $order->get_items();
+
+    // Country, post code, and items are required to calculate the shipping.
+    if (!$country || !$postcode || !$order_items) {
+      return FALSE;
+    }
+
+    $customer = new \WC_Customer();
+    $cart = new \WC_Cart();
+    $shipping = new \WC_Shipping();
+
+    $customer->set_billing_location($country, $state, $postcode, $city);
+    $customer->set_shipping_location($country, $state, $postcode, $city);
+
+    // Add all items to cart.
+    foreach ($order_items as $order_item) {
+      $cart->add_to_cart($order_item->get_product_id(), $order_item->get_quantity());
+    }
+
+    // Calculate shipping.
+    $packages = $cart->get_shipping_packages();
+    $shipping->calculate_shipping($packages);
+    $available_methods = $shipping->get_packages();
+
+    if ($available_methods) {
+      self::$firstAvailableMethod[$order->get_id()] = reset($available_methods[0]['rates']);
+    }
+
+    return self::$firstAvailableMethod[$order->get_id()];
```

Comment:
Triggers a PHP notice/warning if there are no available methods.

```diff
    if ($available_methods) {
      self::$firstAvailableMethod[$order->get_id()] = reset($available_methods[0]['rates']);
    }
+   else {
+     return FALSE;
+   }

    return self::$firstAvailableMethod[$order->get_id()];
```

Or if the calculated result should be sticky even if there are no available methods:
```diff
    if ($available_methods) {
      self::$firstAvailableMethod[$order->get_id()] = reset($available_methods[0]['rates']);
    }
+   else {
+     self::$firstAvailableMethod[$order->get_id()] = FALSE;
+   }

    return self::$firstAvailableMethod[$order->get_id()];
```